commit 1e7a8181640a620300e98e22223ca3445b349840
Author: Vlastimil Babka <vbabka@suse.cz>
Date:   Tue Mar 22 14:44:53 2022 -0700

    mm, fault-injection: declare should_fail_alloc_page()
    
    The mm/ directory can almost fully be built with W=1, which would help
    in local development.  One remaining issue is missing prototype for
    should_fail_alloc_page().  Thus add it next to the should_failslab()
    prototype.
    
    Note the previous attempt by commit f7173090033c ("mm/page_alloc: make
    should_fail_alloc_page() static") had to be reverted by commit
    54aa386661fe as it caused an unresolved symbol error with
    CONFIG_DEBUG_INFO_BTF=y
    
    Link: https://lkml.kernel.org/r/20220314165724.16071-1-vbabka@suse.cz
    Signed-off-by: Vlastimil Babka <vbabka@suse.cz>
    Cc: Mel Gorman <mgorman@techsingularity.net>
    Cc: Matthew Wilcox <willy@infradead.org>
    Cc: David Hildenbrand <david@redhat.com>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

diff --git a/include/linux/fault-inject.h b/include/linux/fault-inject.h
index e525f6957c49..2d04f6448cde 100644
--- a/include/linux/fault-inject.h
+++ b/include/linux/fault-inject.h
@@ -64,6 +64,8 @@ static inline struct dentry *fault_create_debugfs_attr(const char *name,
 
 struct kmem_cache;
 
+bool should_fail_alloc_page(gfp_t gfp_mask, unsigned int order);
+
 int should_failslab(struct kmem_cache *s, gfp_t gfpflags);
 #ifdef CONFIG_FAILSLAB
 extern bool __should_failslab(struct kmem_cache *s, gfp_t gfpflags);